pennylane/labs/vibrational/bosonic.py

```diff
@@ -0,0 +1,615 @@
+
+    def __rsub__(self, other):
+        """Subtract a BoseWord to a constant, i.e. `2 - BoseWord({...})`"""
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot subtract a BoseWord from {type(other)}.")
+
+        if qml.math.size(other) > 1:
+            raise ValueError(
+                f"Arithmetic Bose operations can only accept an array of length 1, "
+                f"but received {other} of length {len(other)}"
+            )
+        self_bs = BoseSentence({self: -1.0})
+        other_bs = BoseSentence({BoseWord({}): other})
+        return self_bs + other_bs
+
+    def __mul__(self, other):
+        r"""Multiply a BoseWord with another BoseWord, a BoseSentence, or a constant.
+
+        >>> w = BoseWord({(0, 0) : '+', (1, 1) : '-'})
+        >>> w * w
+        b⁺(0) b(1) b⁺(0) b(1)
+        """
+
+        if isinstance(other, BoseWord):
+            if len(self) == 0:
+                return copy(other)
+
+            if len(other) == 0:
+                return copy(self)
+
+            order_final = [i[0] + len(self) for i in other.sorted_dic.keys()]
+            other_wires = [i[1] for i in other.sorted_dic.keys()]
+
+            dict_other = dict(
+                zip(
+                    [(order_idx, other_wires[i]) for i, order_idx in enumerate(order_final)],
+                    other.values(),
+                )
+            )
+            dict_self = dict(zip(self.keys(), self.values()))
+
+            dict_self.update(dict_other)
+
+            return BoseWord(dict_self)
+
+        if isinstance(other, BoseSentence):
+            return BoseSentence({self: 1}) * other
+
+        if not isinstance(other, TensorLike):
+            raise TypeError(f"Cannot multiply BoseWord by {type(other)}.")
+
+        if qml.math.size(other) > 1:
+            raise ValueError(
+                f"Arithmetic Bose operations can only accept an array of length 1, "
+                f"but received {other} of length {len(other)}"
+            )
+
+        return BoseSentence({self: other})
+
+    def __rmul__(self, other):
+        r"""Reverse multiply a BoseWord
+
+        Multiplies a BoseWord "from the left" with an object that can't be modified
+        to support __mul__ for BoseWord. Will be defaulted in for example
+        ``2 * BoseWord({(0, 0): "+"})``, where the ``__mul__`` operator on an integer
+        will fail to multiply with a BoseWord"""
+
+        return self.__mul__(other)
+
+    def __pow__(self, value):
+        r"""Exponentiate a Bose word to an integer power.
+
+        >>> w = BoseWord({(0, 0) : '+', (1, 1) : '-'})
+        >>> w**3
+        b⁺(0) b(1) b⁺(0) b(1) b⁺(0) b(1)
+        """
+
+        if value < 0 or not isinstance(value, int):
+            raise ValueError("The exponent must be a positive integer.")
+
+        operator = BoseWord({})
+
+        for _ in range(value):
+            operator *= self
+
+        return operator
+
+    def normal_order(self):
```

Comment:
Using `cform=True` allows creating the following `BoseWord` but applying `normal_order` "corrects" it. What is the main motivation for doing that? Note that in Fermi objects we allow having creation and annihilation operators on the same orbital.